src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,178 @@
+import os
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from torch.functional import Tensor
+
+import requests
+
+from ..feature_extraction_utils import PreTrainedFeatureExtractor
+from ..file_utils import add_end_docstrings, is_torch_available, is_vision_available, requires_backends
+from ..utils import logging
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+if is_vision_available():
+    from PIL import Image
+
+if is_torch_available():
+    import torch
+
+    from ..models.auto.modeling_auto import MODEL_FOR_OBJECT_DETECTION_MAPPING
+
+logger = logging.get_logger(__name__)
+
+
+Prediction = Dict[str, Any]
+Predictions = List[Prediction]
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class ObjectDetectionPipeline(Pipeline):
+    """
+    Object detection pipeline using any :obj:`AutoModelForObjectDetection`. This pipeline predicts bounding boxes of
+    objects and their classes.
+
+    This object detection pipeline can currently be loaded from :func:`~transformers.pipeline` using the following task
+    identifier: :obj:`"object-detection"`.
+
+    See the list of available models on `huggingface.co/models
+    <https://huggingface.co/models?filter=object-detection>`__.
+    """
+
+    def __init__(
+        self,
+        model: "PreTrainedModel",
+        feature_extractor: PreTrainedFeatureExtractor,
+        framework: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__(model, feature_extractor=feature_extractor, framework=framework, **kwargs)
+
+        if self.framework == "tf":
+            raise ValueError(f"The {self.__class__} is only available in PyTorch.")
+
+        requires_backends(self, "timm")
```

Comment:
This is probably only a requirement for DETR. Other object detection models in the future might not rely on timm.

Reply:
@NielsRogge what would you suggest we do here?
add DETR specific check? `if model == detr, then requires_backends(self, "timm")`?

Reply:
Hmm but users should pass a model themselves at initialization? Then that check is already done?

Or is this to load the default object detection model of the pipeline?

Reply:
Currently, the default model is detr. But as you've said:
> model themselves at initialization? Then that check is already done?
Therefore, I should just remove `requires_backends(self, "timm")`. Right?

Basically, `requires_backends(self, "timm")` should check timm on default model initialization, which is detr.